### PR TITLE
Interpolate field on particles

### PR DESCRIPTION
**Pull request by blaisb**
<s>This pull request has a dependency (slight) on #9076. Consequently it might be better to look at this request after #9076 is merged.</s>

In this pull request, we add an additional utilities function to the particles library that allows one to interpolate a field (e..g a TrilinosWrappers::Vector<double>) to the position of the particles into a new field (e.g. a second TrilinosWrappers::Vector<double>). Considering the complexity of the Input-Vector / Output-Vector possible combinations, this function currently only resides in the .h (but if somebody has an inst suggestion...). A unit test for this function is also added.

As always, all your comments are more than welcomed and are highly appreciated :)!.

### CODE REVIEW

**Comment by luca-heltai** [comment]
If anyone wants to take a stab at this, the last three commits are the relevant ones. :)

**Comment by luca-heltai** [comment]
@blaisb , can you rebase, now that the other PR has been merged?

**Comment by blaisb** [comment]
Rebased. This should be ready for review if it passes all tests :)!

**Comment by luca-heltai** [comment]
@blaisb : doxygen complains about this:

~~~

/home/travis/build/dealii/dealii/include/deal.II/particles/utilities.h:196: warning: The following parameters of Particles::Utilities::interpolate_field_on_particles(const DoFHandler< dim, spacedim > &field_dh, const Particles::ParticleHandler< dim, spacedim > &particle_handler, const InputVectorType &field_vector, OutputVectorType &interpolated_field, const ComponentMask &field_comps=ComponentMask()) are not documented:
  parameter 'field_dh'
The command "cat doxygen.log && ! [ -s doxygen.log ]" exited with 1.
~~~

**Comment by blaisb** [comment]
> @blaisb : doxygen complains about this:
> 
> ```
> 
> /home/travis/build/dealii/dealii/include/deal.II/particles/utilities.h:196: warning: The following parameters of Particles::Utilities::interpolate_field_on_particles(const DoFHandler< dim, spacedim > &field_dh, const Particles::ParticleHandler< dim, spacedim > &particle_handler, const InputVectorType &field_vector, OutputVectorType &interpolated_field, const ComponentMask &field_comps=ComponentMask()) are not documented:
>   parameter 'field_dh'
> The command "cat doxygen.log && ! [ -s doxygen.log ]" exited with 1.
> ```

Should be fixed now
:)!

File: source/particles/utilities.cc [hunk 1]
```diff
@@ -209,6 +209,7 @@ namespace Particles
       matrix.compress(VectorOperation::add);
     }
 
+
```
[the remark this answers is not in the document]
Comment (blaisb) [reply source/particles/utilities.cc:212]: I would think we do not need it indeed :)!.

**Comment by blaisb** [comment]
@kronbichler Hopefully this applies correctly all of your suggestions. Thank you very much for taking the time to review. If there is any more necessary changes just ping me and I'll take care of it tomorrow :)!

**Pull request merged**